storage/rocksdb/ha_rocksdb.cc

```diff
@@ -3190,6 +3268,18 @@ class Rdb_transaction {
   virtual void release_lock(const Rdb_key_def &key_descr,
                             const std::string &rowkey, bool force = false) = 0;
 
+  virtual rocksdb::Status lock_range(rocksdb::ColumnFamilyHandle *const cf,
+                                     const rocksdb::Endpoint &start,
+                                     const rocksdb::Endpoint &end) = 0;
+
+  rocksdb::Status lock_singlepoint_range(rocksdb::ColumnFamilyHandle *const cf,
+                                         const rocksdb::Slice &point) {
+    // Normally, one needs to "flip" the endpoint type for reverse-ordered CFs.
+    // But here we are locking just one point so this is not necessary.
```

Comment:
I wonder if this comment is necessarily true. For rev cfs, the locking iterator will always lock adjacent keys using `inf_suffix=true.` However, locking using `inf_suffix=false` here would mean that the locking iterator would not block modifications from happening.

For example, if the locking iterator iterates from 6->8 and then it locks the range `(8, inf_suffix=true)` to `(6, inf_suffix=true)`. A second query could delete 8 since it would use the point lock `(8, inf_suffix=false)` which falls outside of the locked range.

The following shows this is possible:
```
CONNECTION 1:
mysql> show create table t;
+-------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                                                                                                      |
+-------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| t     | CREATE TABLE `t` (
  `i` int NOT NULL,
  `j` int NOT NULL,
  PRIMARY KEY (`i`) COMMENT 'rev:cf',
  KEY `j` (`j`) COMMENT 'cf'
) ENGINE=ROCKSDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci |
+-------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
mysql> begin; select * from t force index (PRIMARY) where i > 5 order by i asc limit 3 for update ;
Query OK, 0 rows affected (0.00 sec)
+---+---+
| i | j |
+---+---+
| 6 | 6 |
| 7 | 7 |
| 8 | 8 |
+---+---+
3 rows in set (2.60 sec)
mysql> select * From information_schema.rocksdb_locks;
+------------------+-----------------+---------------------------------------+------+
| COLUMN_FAMILY_ID | TRANSACTION_ID  | KEY                                   | MODE |
+------------------+-----------------+---------------------------------------+------+
|                3 | 140534119312896 | 0000010780000008:1-0000010780000006:1 | X    |
+------------------+-----------------+---------------------------------------+------+
1 row in set (0.00 sec)
CONNECTION 2:
mysql> delete from t where i = 8;
Query OK, 1 row affected (6.58 sec)
CONNECTION 1:
mysql> select * from t force index (PRIMARY) where i > 5 order by i asc limit 3 for update ;
+---+---+
| i | j |
+---+---+
| 6 | 6 |
| 7 | 7 |
| 9 | 9 |
+---+---+
3 rows in set (3.21 sec)
```

Incidentally, `RangeLockManagerBase::TryLock` also has the same behaviour of always setting `inf_suffix=false` and it's possible that also introduces similar bugs (This is called when we call `get_for_update`, I wonder if we should call `get_for_update` with nullptr as value instead of adding a `lock_singlepoint_range` function here).